package/src/components/MessageInput/MessageInput.tsx

```diff
@@ -325,12 +328,17 @@ const MessageInputWithContext = <
         });
         setSelectedImages(updatedSelectedImages);
       } else if (imageUploadsLength > selectedImagesLength) {
+        /**
+         * User is editing some message which contains image attachments OR
+         * image attachment is added from custom image picker (other than the default bottomsheet image picker)
+         * using `uploadNewImage` function from `MessageInputContext`.
+         **/
         setSelectedImages(
           imageUploads
             .map((imageUpload) => ({
               height: imageUpload.file.height,
               source: imageUpload.file.source,
-              uri: imageUpload.url,
+              uri: imageUpload.url || imageUpload.file.uri,
```

Comment:
if the image is coming from some custom image picker, then `imageUpload.url` will be undefined since the image is still being uploaded to CDN. In that case we need to use the local uri of the image.